cpu/sam0_common/periph/spi.c

```diff
@@ -126,21 +126,16 @@ int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
     /* power on the device */
     poweron(bus);
 
-    /* configuration did not change but device is powered down */
-    if (dev(bus)->BAUD.reg == baud && dev(bus)->CTRLA.reg == ctrla) {
-        /* enable the device */
-        dev(bus)->CTRLA.reg |= SERCOM_SPI_CTRLA_ENABLE;
-
-        return SPI_OK;
+    /* first configuration or reconfiguration after altered device usage */
+    if (dev(bus)->BAUD.reg != baud || dev(bus)->CTRLA.reg != ctrla) {
+        /* disable the device */
+        dev(bus)->CTRLA.reg &= ~(SERCOM_SPI_CTRLA_ENABLE);
+        while (dev(bus)->SYNCBUSY.reg & SERCOM_SPI_SYNCBUSY_ENABLE) {}
+
+        dev(bus)->BAUD.reg = baud;
+        dev(bus)->CTRLA.reg = ctrla;
     }
 
-    /* disable the device */
-    dev(bus)->CTRLA.reg &= ~(SERCOM_SPI_CTRLA_ENABLE);
-    while (dev(bus)->SYNCBUSY.reg & SERCOM_SPI_SYNCBUSY_ENABLE) {}
-
-    dev(bus)->BAUD.reg = baud;
-    dev(bus)->CTRLA.reg = ctrla;
-
     /* also no synchronization needed here, as CTRLA is write-synchronized */
```

Comment:
Just realized this. Should we remove this comment or move it inside the if condition? It might be somewhat confusing to readers because of the synchronization on enable below.

Reply:
IMHO it is better to reword it a bit and move it inside the if.

Reply:
done